### ISSUE TITLE
Socket.Poll timeout occurs unexpectedly

### ISSUE THREAD

**Issue by luigiberrettini** [dotnet/runtime#26209]
I am trying to detect disconnection issues by using `Socket.Poll`.

[Microsoft Docs](https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.socket.poll?view=netcore-2.0) say that `Socket.Poll`:
 - will block execution until the specified time period, measured in microseconds, elapses
 - cannot detect certain kinds of connection problems, such as a broken network cable,
or that the remote host was shut down ungracefully (you must attempt to send or receive data to detect these kinds of errors)
 - when used with `SelectMode.SelectRead` returns:
   - `true` if Listen(Int32) has been called and a connection is pending
   - `true` if data is available for reading
   - `true` if the connection has been closed, reset, or terminated
   - `false` otherwise

Is it possible to detect disconnections with both `SelectMode.SelectWrite` and sending a 0 byte array or something similar that will be ignored by a standard Syslog server that is up and running with the client connected to it?

When the server is up and the connection ok, if I use Poll with `SelectMode.SelectRead` before sending data, I experience a delay due to the timeout expiring.
I do not understand if the expected behavior of Poll is:
 - returning `true` quickly, i.e. no timeout expiration
 - returning `false` only after the timeout elapses

If this is the case to check that everything is ok I should set a very low timeout.
What is a reasonable value i.e. 0 or 1 are ok?

**Comment by karelz** [comment]
This seems answered + no activity for 1.5 years, closing. If there is still some question remaining, I recommend to open a new issue and clarify slearly what is the question. Thanks!
